### ISSUE TITLE
Semantic Highlighting: Some tokens within rust attributes are missing "attribute" modifier.

### ISSUE THREAD

**Issue by Greatness7** [rust-analyzer/rust-analyzer#6536]
I'm loving the *Semantic Highlighting* feature, but it makes quite a mess of (macro/derive/etc) attributes. I suspect reliable highlighting within attributes is not feasible, but at least marking all encompassed tokens with the "attribute" modifier would allow users to override the colors of the worst offenders. (e.g. through configuring `"*.attribute" : "#XXXXXX"` color settings)

An example featuring attributes from the [smart-default](https://crates.io/crates/smart-default) crate:

![example](https://user-images.githubusercontent.com/12927912/98939321-3fa00980-24b7-11eb-99d7-8f5b7b4eacff.PNG)

This crate allows specifying default values with arbitrary expressions passed via attributes. With *Semantic Hightlighting* enabled there are numerous problems with the token handling of said expressions. For one, enums/operators/structs/etc are not recognized at all. For other token types, such as number/punctuation, they do not receive the "attribute" modifier, and thus are not contextually-customizable.

Another example from the [native-windows-gui](https://crates.io/crates/native-windows-gui) crate.
![nwg](https://user-images.githubusercontent.com/12927912/98938295-a7edeb80-24b5-11eb-8616-5c8ed8a302fa.PNG)

I personally find the random/inconsistent highlighting within attributes to be distracting and unpleasant. If it were possible, on my end I would simply override all attribute token colors with a single value, but without consistent modifier presence it cannot be done.

TL;DR
I would love if rust-analyzer's semantic highlighting feature tagged all tokens enclosed within `#[...]` or `#![...]` brackets with the "attribute" modifier.
